src/main/java/org/squiddev/plethora/integration/xnet/MethodsController.java

```diff
@@ -0,0 +1,152 @@
+	 *
+	 */
+
+	/*
+	 *Methods defined in the OC XNet Driver:
+	 * getConnectedBlocks
+	 * getSupportedCapabilities(pos[, side])
+	 * getItems(pos[, side])
+	 * getFluids(pos[, side])
+	 * getEnergy(pos[, side])
+	 * transferItem(sourcePos, sourceSlot, amount, targetPos[, sourceSide[, targetSide]])
+	 * transferFluids(sourcePos, amount, targetPos[, fluidName][, sourceSide[, targetSide]])
+	 * transferEnergy(sourcePos, amount, targetPos[, sourceSide[, targetSide]])
+	 * store(sourcePos, sourceSlot, database, entry[, sourceSide]) - Stores an itemstack in an OC database upgrade
+	 *
+	 *Additional data that we may expose, either method or meta
+	 * NetworkId
+	 * Channels
+	 * RF storage - should already be exposed
+	 * Logic state - the colors are defined in mcjty.xnet.api.channels.Color,
+	 *   usage shown in mcjty.xnet.api.helper.AbstractConnectorSettings.calculateColorsMask
+	 *   Alternatively, a player could set a Redstone Proxy and read the state;
+	 *   we could theoretically expose a
+	 */
+
+
+	//----------
+	//Start OC XNet Driver stubs
+	//----------
+	// This block is intended to simplify adapting the code from the OC XNet Driver mod,
+	// and may be removed at a later time.
+
+	//TODO Do we need this?  This was a configuration setting in OC XNet Driver
+	private static final boolean relativePositions = false;
+
+	//TODO File an issue with OC XNet Driver; they had the same check for `toRelative` and `toAbsolute`
+	private static BlockPos toRelative(BlockPos pos, BlockPos controllerPos) {
+		return relativePositions
+			? pos
+			: pos.add(-controllerPos.getX(), -controllerPos.getY(), -controllerPos.getZ());
+	}
+
+	private static BlockPos toAbsolute(BlockPos pos, BlockPos controllerPos) {
+		return !relativePositions
+			? pos
+			: pos.add(controllerPos.getX(), controllerPos.getY(), controllerPos.getZ());
+	}
+
+	@Nullable
+	private static SidedPos getSidedPos(BlockPos pos, @Nonnull IControllerContext context) {
+		return context.getConnectedBlockPositions().stream()
+			.filter(sp -> sp.getPos().equals(pos)).findFirst()
+			.orElse(null);
+	}
+
+	//----------
+	//End OC XNet Driver stubs
+	//----------
+
+	//TODO Check if this (and the other methods in this class)
+	// can use IControllerContext instead of TileEntityController
+	@PlethoraMethod(
+		modId = XNet.MODID,
+		doc = "-- List all blocks connected to the XNet network"
+	)
+	public static Map<Integer, Object> getConnectedBlocks(@Nonnull IContext<TileEntityController> context) {
+		//REFINE Do we want to return a Tuple<boolean, ?> representing the success and result/error?
+		LuaList<Object> out = new LuaList<>();
+		TileEntityController controller = context.getTarget();
+
+		IWorldLocation location = context.getContext(ContextKeys.ORIGIN, IWorldLocation.class);
+		if (location == null) return Collections.emptyMap(); //REFINE Do I need to return an empty map for a method?
```

Comment:
In this particular case, do:

```java
getConnectedBlocks(
  @FromTarget TileEntityController controller,
  @FromContext(ContextKeys.ORIGIN) IWorldLocation location
) {
```

Basically means the method is only exposed if we've got a location, and does all the injection for you. 

My general recommendation is to try to avoid using `IContext` as much as possible in methods (within reason, of course), as it's entirely dynamic. Doesn't make much difference now, but might if I ever get round to my oft-dreamed-of refactors :).

Reply:
Yeah, I need to dig into the various 'advanced' args for `PlethoraMethod`s; heck, I left a memo comment in the template for them!